pkgs/servers/mail/mailhog/default.nix

```diff
@@ -1,19 +1,23 @@
-{ lib, buildGoPackage, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "MailHog";
   version = "1.0.1";
 
-  goPackagePath = "github.com/mailhog/MailHog";
-
   src = fetchFromGitHub {
     owner = "mailhog";
     repo = "MailHog";
     rev = "v${version}";
-    sha256 = "124216850572r1h0ii7ad6jd1cd5czcvkz7k2jzvjb4pv2kl8p3y";
+    hash = "sha256-flxEp9iXLLm/FPP8udlnpbHQpGnqxAhgyOIUUJAJgog=";
   };
 
-  ldflags = [ "-s" "-w" "-X main.version=${version}" ];
+  postPatch = ''
+    go mod init github.com/mailhog/MailHog
+  '';
```

Comment:
Why this?

Reply:
The upstream project doesn't use Go modules. To enable building the project with `buildGoModule`, we initialize a module before the build.